grpc.go

```diff
@@ -145,19 +158,55 @@ func onlineMemResources() error {
 	return err
 }
 
-func onlineCPUMem(req *pb.OnlineCPUMemRequest) error {
+func (a *agentGRPC) onlineCPUMem(req *pb.OnlineCPUMemRequest) error {
 	if req.NbCpus <= 0 {
-		return fmt.Errorf("requested number of CPUs '%d' must be greater than 0", req.NbCpus)
+		return handleError(req.Wait, fmt.Errorf("requested number of CPUs '%d' must be greater than 0", req.NbCpus))
 	}
 
 	onlineCPUMemLock.Lock()
 	defer onlineCPUMemLock.Unlock()
 
 	if err := onlineCPUResources(req.NbCpus); err != nil {
-		return err
+		return handleError(req.Wait, err)
+	}
+
+	if err := onlineMemResources(); err != nil {
+		return handleError(req.Wait, err)
+	}
+
+	// At this point all CPUs have been connected, we need to know
+	// the actual range of CPUs
+	cpus, err := ioutil.ReadFile(sysfsConnectedCPUsPath)
+	if err != nil {
+		return handleError(req.Wait, fmt.Errorf("Could not get the actual range of connected CPUs: %v", err))
+	}
+	connectedCpus := strings.Trim(string(cpus), "\t\n ")
+
+	// In order to update container's cpuset cgroups, docker's cpuset cgroup MUST BE updated with
+	// the actual number of connected CPUs
+	if err := ioutil.WriteFile(sysfsDockerCpusetPath, []byte(connectedCpus), dockerCpusetMode); err != nil {
+		return handleError(req.Wait, fmt.Errorf("Could not update docker cpuset cgroup '%s': %v", connectedCpus, err))
 	}
 
-	return onlineMemResources()
+	// Now that we know the actual range of connected CPUs, we need to iterate over
+	// all containers an update each cpuset cgroup. This is not required in docker
+	// containers since they don't hot add/remove CPUs.
+	for _, c := range a.sandbox.containers {
```

Comment:
why should be updated on each container in the sandbox, the cpu update is not an operation per container ? In that case we should receive the container that will be updated, isn't it?

Reply:
nop, in the case of a POD we can have several containers, when new CPUs are added, these new resources need to be visible for all containers

Reply:
thanks.